app/views/layouts/application.html.erb

```diff
@@ -108,7 +108,9 @@
       </form>
     <% end %>
 
-    <div class="hidden" id="data-preloaded" data-preloaded="<%= preloaded_json %>"></div>
+    <script nonce="<%= csp_nonce %>">
```

Comment:
What happens if you use `script type="text/discourse-preload"` instead?

Reply:
That even bypasses the need for a nonce! Which also fixed the perf concern from @eviltrout  and the nonce cache from @davidtaylorhq

Reply:
On the other hand means I need to escape script close tags :thinking: